kale/pipeline/mpca_trainer.py

```diff
@@ -0,0 +1,159 @@
+# =============================================================================
+# Author: Shuo Zhou, szhou20@sheffield.ac.uk
+#         Haiping Lu, h.lu@sheffield.ac.uk or hplu@ieee.org
+# =============================================================================
+
+"""Implementation of MPCA->Feature Selection->Linear SVM/LogisticRegression Pipeline
+"""
+
+import logging
+
+import numpy as np
+from sklearn.base import BaseEstimator, ClassifierMixin
+from sklearn.feature_selection import f_classif
+from sklearn.linear_model import LogisticRegression
+from sklearn.model_selection import GridSearchCV
+from sklearn.svm import SVC
```

Comment:
Have you considered [`LinearSVC` (liblinear)](https://scikit-learn.org/stable/modules/generated/sklearn.svm.LinearSVC.html)? Or, can we make it an option (with one as the default)? SVC is using libsvm but LinearSVC is using liblinear.

I noticed this when checking https://scikit-learn.org/stable/modules/feature_selection.html and found that the examples using `LinearSVC` rather than `SVC`.

Did you check the matlab SVM I sent to you to see which version was used?

Reply:
SVC object has predict_proba function, which can give the probbabilitiy of each class, while LinearSVC does not have it.

Reply:
**If** LinearSVC can give a better accuracy and the user does not care the probability, then it is still a decent choice. How often did we report the probability of each class in our papers so far?

I understand probability is something good, but I do not consider it as essential or must-have.

Reply:
Probability is a feature requested by Cameron. How about making it optional (`LinearSVC` and `SVC(kernel="linear")`)?

Reply:
A feature requested by a user does not mean to enforce it for all users. Otherwise, sklearn won't have LinearSVC. Good to have options.